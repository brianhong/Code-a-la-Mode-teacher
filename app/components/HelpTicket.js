// @flow
import React from 'react';
import type { ticketType } from '../actions/tickets';

const HelpTicket = ({ ticket }: {ticket: ticketType}) => (
  <div>
    <pre>{`${ticket.question}`}</pre>
  </div>
);

export default HelpTicket;
