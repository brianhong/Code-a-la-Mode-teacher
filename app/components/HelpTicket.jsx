// @flow
import React from 'react';
import Flexbox from 'flexbox-react';

import FlatButton from 'material-ui/FlatButton';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import type { ticketType } from '../actions/tickets-actions';

// just a skeleton, not sure how tickets will actually be represented yet
const truncate = (question: string) => `${question.slice(0, 15)}...`;

const HelpTicket = ({ ticket }: {ticket: ticketType}) => (
  <Card>
    <CardHeader
      title={ticket.question.length > 15 ? truncate(ticket.question) : ticket.question}
      actAsExpander={true}
      showExpandableButton={true}
    />
    {
      ticket.question.length > 15
      ? <CardText expandable={true}>
        {ticket.question}
      </CardText>
      : null
    }
    <CardActions expandable={true}>
      <Flexbox justifyContent="center">
        <FlatButton label="Show Code" />
        <FlatButton label="Close" />
      </Flexbox>
    </CardActions>
  </Card>
);

export default HelpTicket;
