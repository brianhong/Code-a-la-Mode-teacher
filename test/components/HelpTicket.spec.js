import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HelpTicket from '../../app/components/HelpTicket';

describe('HelpTicket component', () => {
  const testTicket1 = { id: 1, question: 'is this question being rendered?' };
  const component = shallow(<HelpTicket ticket={testTicket1} />);

  it('should receive a ticket object in props', () => {
    expect(component.instance().props.ticket).toBe(testTicket1);
  });

  it('should display the a ticket question', () => {
    const tree = renderer
      .create(
        <div>
          <pre>{`${testTicket1.question}`}</pre>
        </div>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
