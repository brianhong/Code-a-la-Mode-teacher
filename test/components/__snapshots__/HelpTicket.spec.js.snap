// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpTicket component should display the a ticket question 1`] = `
<div>
  <pre>
    is this question being rendered?
  </pre>
</div>
`;
